pkg/infra/baseAgent.go

```diff
@@ -262,75 +265,42 @@ func (a *Base) UpdateHPChange(change int) {
 	}
 }
 
-func (a *Base) utility() float64 {
-	// TODO: Improve utility calculation
-	// Currently just use ratio between food taken and food seen
-	// Somehow also factor
-	// Ideas: piecewise/bounded function with log, concave quadratic
-
-	hpGainFactor := netHPGainFactor(a.totalHPGained, a.totalHPLost, a.tower.healthInfo.MaxHP)
-	foodIntakeFactor := foodIntakeFactor(a.totalFoodTaken, a.totalFoodSeen)
-	// this means the multiplcation of these values can be larger than 1
-	return simpleQuadraticUtility(hpGainFactor * foodIntakeFactor)
-}
-
-/*
-HP Gain Factor.
-inputs:
-- gain >= 0
-- loss >= 0
-
-output range:
-- y >= 0.
-
-key values:
-- = 1: agent is sustaining overall
-- < 1: agent is losing hp on average
-- > 1: agent gaining hp on average
-- when loss is 0, default to percentage difference between gain and loss. Acknowledge that HP gain should increase utility
-*/
-// func hpGainFactor(gain, loss, maxhp int) float64 {
-// 	if loss == 0 {
-// 		return 1 + float64(gain)/float64(maxhp)
-// 	}
-// 	return float64(gain) / float64(loss)
-// }
-
-/*
-Net HP Gain Factor.
-This one uses net hp gain as opposed to hp gain/loss ratio.
-*/
-func netHPGainFactor(gain, loss, maxhp int) float64 {
-	netHPGain := gain - loss
-	multiplier := 1 + float64(netHPGain)/float64(maxhp) // reward positive net hp gain, punish otherwise
-	return utilFunctions.RestrictToRange(0, math.Inf(1), multiplier)
-}
-
-/*
-Food Intake Factor.
-inputs:
-- intake >= 0
-- total >= 0
-range: [0, 1]. == 1, agent has taken all the food it has seen
-
-*/
-func foodIntakeFactor(intake, seen food.FoodType) float64 {
-	// If an agent hasn't seen any food, has 0 utility
-	if seen == 0 {
-		return 0
+// Updates the utility of a given agent
+// For this paper, we take N agents i ∈{1,...,N}that perform the following actions in each iterated round t ∈{1,...,∞}:
+// Determines the resources it has available, gi ∈[0,1]
+// Determines its need for resources, qi ∈[0,1]
+// Makes a provision of resources, pi ∈[0,1]
+// Makes a demand for resources, di ∈[0,1]
+// Receives an allocation of resources, ri ∈[0,1]
+// Makes an appropriation of resources, r′i ∈[0,1]
+// The total resources accrued at the end of a round is hence given by:
+// Ri = r′i + (gi −pi) (1)
+// Which provides the utility of an agent, ui, as:
+// ui =
+// αi(qi) + βi(Ri −qi), if Ri ≥qi
+// αi(Ri) −γi(qi −Ri), otherwise
+func (a *Base) utility(foodRequested, foodTaken food.FoodType) {
+	alpha := 0.2
+	beta := 0.1
+	gamma := 0.18
+
+	g := float64(a.CurrPlatFood()) / float64(a.tower.maxPlatFood)
+	var q float64
+	if a.daysAtCritical == a.HealthInfo().MaxDayCritical {
+		q = 1.0
+	} else {
+		q = 0.0
 	}
```

Comment:
```suggestion
	resNeed := 0.0
	if a.daysAtCritical == a.HealthInfo().MaxDayCritical {
		resNeed = 1.0
	}
```

Reply:
Fair.

Since this formula comes direct from literature (and is explicitly defined above), I would rather keep the variable names as letters

Reply:
It's more readable this way - report can use the letters, but here it's quite obscure.

Reply:
If we use specific variable names in the report, we should probably use the same ones in the code if possible (especially if the formulas are taken from literature). I would rather keep it as it is

Reply:
I agree with Jason. Formulas define the concept, but it's better to set readable variable names in the code so the logic can be easily evaluated. E.g: if `q` represents allocation in the formula, it is easier to use a variable name like `foodAlloc` rather than `q`, so when someone is reviewing the code, they can quickly go "aha,`foodAlloc` is being computed correctly" or "hmm I think `foodAlloc` can be obtained in a better way"

Reply:
We can do it this way :)

I'll push with better variables